packages/core/tests/helpers.ts

```diff
@@ -818,3 +820,55 @@ export async function setupProofsTest(faberName: string, aliceName: string, auto
     aliceReplay,
   }
 }
+
+export function getKeysFromFormatData(formatData: GetFormatDataReturn<[IndyProofFormat]>): {
+  proposeKey1: string
+  proposeKey2: string
+  requestKey1: string
+  requestKey2: string
+} {
+  const proofRequest: IndyProofRequest | undefined = formatData.request?.indy ?? undefined
+  const proofProposal: IndyProofRequest | undefined = formatData.proposal?.indy ?? undefined
+  let proposeKey1, proposeKey2: string
+
+  if (!proofProposal) {
+    throw new AriesFrameworkError('missing indy proof proposal')
+  }
+  if (!proofRequest) {
+    throw new AriesFrameworkError('missing indy proof request')
+  }
+  const proposeAttributes: Map<string, ProofAttributeInfo> = proofProposal[
+    'requested_attributes' as keyof IndyProofRequest
+  ] as Map<string, ProofAttributeInfo>
+
+  // these keys are dynamically generated from UUID
+
+  // eslint-disable-next-line prefer-const
+  proposeKey1 = Object.keys(proposeAttributes)[1]
+
+  const proposePredicates: Map<string, ProofAttributeInfo> = proofProposal[
+    'requested_predicates' as keyof IndyProofRequest
+  ] as Map<string, ProofAttributeInfo>
+
+  // eslint-disable-next-line prefer-const
+  proposeKey2 = Object.keys(proposePredicates)[0]
+
+  // eslint-disable-next-line prefer-const
+  let requestKey1, requestKey2: string
+
+  const attributes: Map<string, ProofAttributeInfo> = proofRequest[
+    'requested_attributes' as keyof IndyProofRequest
+  ] as Map<string, ProofAttributeInfo>
+
+  // eslint-disable-next-line prefer-const
+  requestKey1 = Object.keys(attributes)[1]
+
+  const predicates: Map<string, ProofAttributeInfo> = proofRequest[
+    'requested_predicates' as keyof IndyProofRequest
+  ] as Map<string, ProofAttributeInfo>
+
+  // eslint-disable-next-line prefer-const
+  requestKey2 = Object.keys(predicates)[0]
+
+  return { proposeKey1, proposeKey2, requestKey1, requestKey2 }
```

Comment:
This can be simplified quite a bit, however I'm not sure if this can go as a generic test util. You access _very_ specific indexes in the proof request, meaning you can't really reuse this in other tests -- only between the v1 and v2 get format data return test. So maybe it should be hosted somewhere in the proof module tests folder. Just want to make sure we're not cluttering this file with methods that aren't actually reusable.

```suggestion
  const proofRequest = formatData.request?.indy
  const proofProposal = formatData.proposal?.indy

  if (!proofProposal) {
    throw new AriesFrameworkError('missing indy proof proposal')
  }
  if (!proofRequest) {
    throw new AriesFrameworkError('missing indy proof request')
  }

  const proposeAttributes = proofProposal.requested_attributes
  const proposePredicates = proofProposal.requested_predicates
  const requestAttributes = proofRequest.requested_attributes
  const requestPredicates = proofRequest.requested_predicates

  const proposeKey1 = Object.keys(proposeAttributes)[1]
  const proposeKey2 = Object.keys(proposePredicates)[0]

  const requestKey1 = Object.keys(requestAttributes)[1]
  const requestKey2 = Object.keys(requestPredicates)[0]

  return { proposeKey1, proposeKey2, requestKey1, requestKey2 }
```